pkg/virt-manifest/rest/manifest.go

```diff
@@ -94,11 +95,13 @@ func (m manifest) mapManifest(_ context.Context, request interface{}) (interface
 
 	virt_manifest.AddMinimalVMSpec(vm)
 
-	mappedVm, err := virt_manifest.MapVM(m.connection, vm)
-	mappedVm.Spec.Domain.Name = mappedVm.ObjectMeta.Name
+	client, err := kubecli.Get()
+
+	mappedVm, err := virt_manifest.MapVM(m.connection, vm, client)
```

Comment:
I don't see a reason to avoid it - it would just needless lead to divergent behaviour between the manifest code & the code that actually runs during launch. As we add support for different resources, we are going to need to resolve this in order to correctly fill out the defaults. For example, if a network device that is to be connected with vhost-user, will have an effect on the virtual hardware choices that are filled in, which can be different from the choices made with macvtap. So by not resolving referenced resources, we are likely to incorrectly fill in defaults in the manifest